pkg/virt-controller/watch/vm.go

```diff
@@ -2644,14 +2637,23 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 			syncErr = &syncErrorImpl{fmt.Errorf("Error encountered while handling CPU change request: %v", err), HotPlugCPUErrorReason}
 		}
 
+		var vmUpdateErr error
 		if syncErr == nil {
 			if !equality.Semantic.DeepEqual(vm, vmCopy) {
-				vm, err = c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy)
-				if err != nil {
-					syncErr = &syncErrorImpl{fmt.Errorf("Error encountered when trying to update vm according to add volume and/or memory dump requests: %v", err), FailedUpdateErrorReason}
+				vm, vmUpdateErr = c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy)
+				if vmUpdateErr != nil {
+					syncErr = &syncErrorImpl{fmt.Errorf("Error encountered when trying to update vm according to add volume and/or memory dump requests: %v", vmUpdateErr), FailedUpdateErrorReason}
 				}
 			}
 		}
+
+		if vmUpdateErr == nil && c.clusterConfig.HotplugNetworkInterfacesEnabled() {
```

Comment:
Is there a specific reason the block is not placed under the condition `if syncErr == nil`?

The result now is still problematic:
- If `syncErr != nil`, the update will not occur and we will enter this new condition and perform a VMI patch although we should not have.

If it is not breaking anything else, please place the block inside that condition:

```go
if syncErr == nil {
  if !equality.Semantic.DeepEqual(vm, vmCopy) {
    vm, err = c.clientset.VirtualMachine(vmCopy.Namespace).Update(context.Background(), vmCopy)
    if err != nil {
      syncErr = ...
    }
  }
    
  if syncErr == nil && c.clusterConfig.HotplugNetworkInterfacesEnabled() {
    ...
  }
}
```

Please try to avoid changing stuff that is not a must, just to express no changes have been required in the current flow/code.
If it bothers you a lot, you can rename the errors in a separate commit.